ls1intum/Artemis
# `General`: Enhance mobile appearance of various charts

### Checklist
#### General


#### Client

### Motivation and Context


Currently, many charts do not look good on smaller screens, e.g. smartphones or tablets. This is reducing the user experience for respective users.
### Description

I improved the appearance of various charts for mobile layouts (at least I tried to). The following views are enhanced by this PR:

- Course details
- Course scores
- Course statistics
- Exam statistics
- Exercise details
- Exercise statistics
- Quiz statistics

Further (smaller) improvements:

- The labels of the vertical bar charts on the course- and exercise statistics containing natural language (e.g. weekdays) are now instant translation sensitive.
- The bar labels of various vertical bar charts now rotate from horizontal to 45 deg if the screen gets smaller
- I removed an unnecessary directive declaration for the quiz statistics
- I removed the possibility to show the active students line chart in fullscreen. The tooltips are not visible in fullscreen. Thus, the fullscreen mode does not add much benefit to the user.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with at least one quiz, one other exercise and an exam with participations

1. Log in to Artemis
2. Navigate to Course Administration
3. Check the views with different screen sizes. Make sure everything is readable and nothing looks distorted.
4. For the course and exercise statistics: Check that the bar labels translate directly if you change the language settings.
5. If you have any suggestions, let me know. I cannot promise to be able to fulfill them but I can try.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| question-statistic.component.ts | 65% | 85.22% |
| quiz-point-statistic.component.ts | 75.43% | 93.1% |
| quiz-statistic.component.ts | 72.22% | 90.9% |
| quiz-statistics.ts | 87.5% | 97.56% |
| participant-scores-distribution.component.ts | 85.48% | 93.96% |
|  statistics-graph.component.ts | 78.26% | 94.31% |

### Screenshots


<details>
<summary>Course details:</summary>

![artemis-test1 artemis in tum de_course-management_7(iPhone XR)](https://user-images.githubusercontent.com/80622272/173117758-8802502a-74a0-473c-9f5a-1c6fd426e50e.png)

</details>

<details>
<summary>Course scores:</summary>

![localhost_9000_course-management_5_scores(iPhone 12 Pro)](https://user-images.githubusercontent.com/80622272/169095590-2e45727e-c298-41ea-9a35-5a6fa0b25a3f.png)
</details>

<details>
<summary>Course statistics:</summary>

![localhost_9000_course-management_5_course-statistics(iPhone 12 Pro)](https://user-images.githubusercontent.com/80622272/169096424-0ed6dd1f-8a22-4071-af02-08623143641f.png)
</details>

<details>
<summary>Exam statistics:</summary>

![localhost_9000_course-management_5_exams_9_scores(iPhone 12 Pro)](https://user-images.githubusercontent.com/80622272/169096765-9517bce8-8b64-4f3b-bc46-29d85cf366ac.png)
</details>

<details>
<summary>Exercise details:</summary>

![localhost_9000_course-management_5_modeling-exercises_75(iPhone 12 Pro)](https://user-images.githubusercontent.com/80622272/169102649-1165054f-dfb1-442d-8642-ce9c78bdd289.png)
</details>

<details>
<summary>Exercise statistics:</summary>

![localhost_9000_course-management_5_text-exercises_79(iPhone 12 Pro)](https://user-images.githubusercontent.com/80622272/169097423-44aa025b-3a98-4a47-acc9-1b0481ef14d7.png)
</details>

<details>
<summary>Quiz statistics:</summary>

![localhost_9000_course-management_5_text-exercises_79(iPhone 12 Pro) (1)](https://user-images.githubusercontent.com/80622272/169097907-a2c7b09c-d9ea-4a27-9b99-28bb5ee8bc57.png)

![localhost_9000_course-management_5_text-exercises_79(iPhone 12 Pro) (2)](https://user-images.githubusercontent.com/80622272/169097989-14dbd39a-17aa-4073-8364-7459ae8be43d.png)

![localhost_9000_course-management_5_text-exercises_79(iPhone 12 Pro) (3)](https://user-images.githubusercontent.com/80622272/169098095-5978bf92-9fe0-4d24-9674-03f6df2b765a.png)

![localhost_9000_course-management_5_quiz-exercises_131_dnd-question-statistic_53(iPhone 12 Pro)](https://user-images.githubusercontent.com/80622272/169100440-18188c53-fd5d-4d06-9eb8-8b711895ca96.png)

![localhost_9000_course-management_5_quiz-exercises_159_sa-question-statistic_60(iPhone XR)](https://user-images.githubusercontent.com/80622272/170475403-770f0e25-1cfc-4e84-9ff6-eebbd8fe1c71.png)

</details>
